OPM/opm-rule-onboarding.ps1

```diff
@@ -0,0 +1,540 @@
+
+$InDebug = $PSBoundParameters.Debug.IsPresent
+$InVerbose = $PSBoundParameters.Verbose.IsPresent
+
+# Global URLS
+# -----------
+$URL_PVWABaseAPI = $PVWAURL+"/WebServices/PIMServices.svc"
+$URL_PVWAAPI = $PVWAURL+"/api"
+$URL_Authentication = $URL_PVWAAPI+"/auth"
+$URL_Logon = $URL_Authentication+"/$AuthType/Logon"
+$URL_Logoff = $URL_Authentication+"/Logoff"
+
+# URL Methods
+# -----------
+$URL_PlatformCommands = $URL_PVWABaseAPI+"/Policy/{0}/PrivilegedCommands"
+
+
+# Script Defaults
+# ---------------
+$g_CsvDefaultPath = $Env:CSIDL_DEFAULT_DOWNLOADS
+
+
+# Initialize Script Variables
+# ---------------------------
+$g_LogonHeader = ""
+
+#region Functions
+Function Test-CommandExists
+{
+    Param ($command)
+    $oldPreference = $ErrorActionPreference
+    $ErrorActionPreference = 'stop'
+    try {if(Get-Command $command){RETURN $true}}
+    Catch {Write-Host "$command does not exist"; RETURN $false}
+    Finally {$ErrorActionPreference=$oldPreference}
+} #end function test-CommandExists
+
+Function Encode-URL($sText)
+{
+	if ($sText.Trim() -ne "")
+	{
+		Log-Msg -Type Debug -Msg "Returning URL Encode of $sText"
+		return [System.Web.HttpUtility]::UrlEncode($sText)
+	}
+	else
+	{
+		return $sText
+	}
+}
+
+Function Get-TrimmedString($sText)
+{
+	if($sText -ne $null)
+	{
+		return $sText.Trim()
+	}
+	# Else
+	return $sText
+}
+
+Function Log-MSG
+{
+<# 
+.SYNOPSIS 
+	Method to log a message on screen and in a log file
+.DESCRIPTION
+	Logging The input Message to the Screen and the Log File. 
+	The Message Type is presented in colours on the screen based on the type
+.PARAMETER MSG
+	The message to log
+.PARAMETER Header
+	Adding a header line before the message
+.PARAMETER SubHeader
+	Adding a Sub header line before the message
+.PARAMETER Footer
+	Adding a footer line after the message
+.PARAMETER Type
+	The type of the message to log (Info, Warning, Error, Debug)
+#>
+	param(
+		[Parameter(Mandatory=$true)]
+		[AllowEmptyString()]
+		[String]$MSG,
+		[Parameter(Mandatory=$false)]
+		[Switch]$Header,
+		[Parameter(Mandatory=$false)]
+		[Switch]$SubHeader,
+		[Parameter(Mandatory=$false)]
+		[Switch]$Footer,
+		[Parameter(Mandatory=$false)]
+		[ValidateSet("Info","Warning","Error","Debug","Verbose")]
+		[String]$type = "Info"
+	)
+	try{
+		If ($Header) {
+			"=======================================" | Out-File -Append -FilePath $LOG_FILE_PATH 
+			Write-Host "======================================="
+		}
+		ElseIf($SubHeader) { 
+			"------------------------------------" | Out-File -Append -FilePath $LOG_FILE_PATH 
+			Write-Host "------------------------------------"
+		}
+	
+		$msgToWrite = "[$(Get-Date -Format "yyyy-MM-dd hh:mm:ss")]`t"
+		$writeToFile = $true
+		# Replace empty message with 'N/A'
+		if([string]::IsNullOrEmpty($Msg)) { $Msg = "N/A" }
+		# Mask Passwords
+		if($Msg -match '((?:"password"|"secret"|"NewCredentials")\s{0,}["\:=]{1,}\s{0,}["]{0,})(?=([\w!@#$%^&*()-\\\/]+))')
```

Comment:
Make sure this regex fits your needs and that nay secret that is beign used or pushed is not visible in the log file

Reply:
Thanks. For this, we don't have any secrets inputted.